### ISSUE TITLE
Add possibility to install locally

### ISSUE THREAD

**Issue by raulfdm** [pedsmoreira/battlecry#5]
Is there a specific reason to only be able to install global?

I think it could be nice to be able to install locally and configure everything inside a project in order to easy share/install everything for all project maintainers.

**Comment by pedsmoreira** [comment]
I agree, it would be nice to be able to install locally.

The package resolution used with babel is quite tricky in this case, as the packages are inside battlecry and not on your project directly.

If you want to create a PR for this issue, please feel free to do so.

**Comment by raulfdm** [comment]
Fine! I'll do it

**Comment by raulfdm** [comment]
@pedsmoreira After few hours of work, I've realize since it's a Node package/cli, what if we use the regular syntax?

```js
// before
import { bar } from './foo'

//after
const { bar } = require('./foo')
```

I've configured [rollup](https://rollupjs.org) to generate a build and I had some issues and it causes some concerns against this major change, but I'll try to explain:

## Reasons
1.  Since battlecry is a Node package, it's a good practice to write in Node syntax
1.  Since we're building up to CommonJS, it generate conflicts https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module
1.  Final bundle will be God damm smaller! (really)
1.  Shorter time to build (since we don't have to run too many plugins/converters)
1.  This text explain much better than me why: https://medium.com/@kelin2025/so-you-wanna-use-es6-modules-714f48b3a953

Of course it'll be up to you accept it or not. I'll continue work on that to generate a awesome version and show you. :)

Cheers man 🍺

**Comment by raulfdm** [comment]
Already start to work on it... When I was almost finish it, I faced a really serious problem: if we transpile the code and generate a final, it'll have some major changes.

That's because the final code will be ES5 and the `battlecry-setup` file accept ES6 modules (you can check the progress [here](https://github.com/raulfdm/battlecry)).

I thought babel-node could be useful for this case, but they [discourage to use it in production](https://babeljs.io/docs/en/next/babel-node.html).

**Comment by raulfdm** [comment]
Note: 

I've tried use `format: 'ems'` but isn't solve the problem, since it uses ES6 import/exports

**Comment by raulfdm** [comment]
Note [2]:

I've realize for this propose the best option is use `@babel/cli` to compile the code and keep the same folder structure.

Everything it's looks work, however babel also compile the internal generators templates. It'll be necessary a workaround for this specific case.

**Comment by raulfdm** [comment]
Note [3]: 

This package isn't kind of compiled. I've try so many different ways but none of them are good enough and will keep compatibility.

Today I've realized about everything, and maybe `@babel/node` will solve this problem. Isn't recommended for production but `battlecry` only runs on dev mode so.. it could be a big deal and save a lot of effort.

**Comment by pedsmoreira** [comment]
Hey @raulfdm, I can see you've been making a lot of progress understanding the code base.

"This package isn't kind of compiled" - that sentence pretty much sums it up I guess.

And yes, `battlecry` is meant for development only, so using an unstable library such as `@babel/node` wouldn't be as big of an issue.

If you decide to go with `babel-node`, it may be worth considering switching to Babel 7 - there is a closed pull request that started doing it, it may be worth taking a look at it: https://github.com/pedsmoreira/battlecry/pull/1

**Comment by raulfdm** [comment]
I've started again from scratch and I think I found the problem to install it locally.

```
ReferenceError: Unknown plugin "/Users/raulmelo/Desktop/test-cry/node_modules/battlecry/bin/../node_modules/babel-plugin-transform-async-to-generator" specified in "base"
```

But pay attention at this URL: `/Users/raulmelo/Desktop/test-cry/node_modules/battlecry/bin/../node_modules/babel-plugin-transform-async-to-generator`

It's considering that inside `battlecry` dependency folder it has `node_modules`.

I fixed it returning: 
```
join(__dirname, `../../babel-${type}-${name}`);
```

and as result we have: `/Users/raulmelo/Desktop/test-cry/node_modules/babel-plugin-transform-async-to-generator`.

However, I'm sure it'll stop to work globally, so, I'm trying to think a way to check if it's being running locally or globally and then pass a correct path.
